token/js/client/token.js

```diff
@@ -2240,9 +2240,6 @@ export class Token {
     mint: PublicKey,
     owner: PublicKey,
   ): Promise<PublicKey> {
-    if (!PublicKey.isOnCurve(owner.toBuffer())) {
```

Comment:
Instead of removing this check, can you gate it on an optional `allow_pda_owner` (or similar) flag that defaults to false?  Unintentionally setting an off-curve owner will result in loss of funds

Reply:
Imho I don't think getAssociatedTokenAddress should take the responsibility, the developer do.
Maybe I separate them into curve-check version and without checking version?
I think the flag cause some smell in the code.
what do you think ?

Reply:
> Imho I don't think getAssociatedTokenAddress should take the responsibility, the developer do.

Relying on the developer to make this check has already resulted in non-trivial losses of user funds

> Maybe I separate them into curve-check version and without checking version?
> I think the flag cause some smell in the code.
> what do you think ?

If you want to hide the flag behind two public methods that call the same private method, that's fine.  `getAssociatedTokenAddress` must continue to do the check by default though.